src/native/windows/apps/prunsrv/prunsrv.c

```diff
@@ -1375,11 +1418,18 @@ static DWORD WINAPI serviceStop(LPVOID lpParameter)
         CloseHandle(gSignalThread);
         gSignalEvent = NULL;
     }
-    if (wait_to_die && !timeout)
-        timeout = 300 * 1000;   /* Use the 5 minute default shutdown */
 
-    if (dwCtrlType == SERVICE_CONTROL_SHUTDOWN)
-        timeout = MIN(timeout, apxGetMaxServiceTimeout(gPool));
+    /* We already waited here timeout is msec */
+    waited = waitedSinceStopCmd();
+    apxLogWrite(APXLOG_MARK_DEBUG "Waited %d timeout (%d)", waited, timeout);
+    if (timeout > waited) {
+        timeout = timeout - waited;
+    } else {
+        /* something is wrong, the timeout is too small */
+        apxLogWrite(APXLOG_MARK_DEBUG "Waiting more than the specified timeout (%d)", timeout);
+    }
```

Comment:
Include how long we have waited in this message?

Reply:
Done.